operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -407,6 +414,130 @@ func (r *ReconcileIstioOperator) Reconcile(_ context.Context, request reconcile.
+	operatorManageWebhooks := operatorManageWebhooks(iop)
+	if !operatorManageWebhooks && (!exists || isDefaultInstallation) {
+		if rev == "" {
+			rev = revtag.DefaultRevisionName
+		}
+		autoInjectNamespaces := validateEnableNamespacesByDefault(iop)
+
+		o := &revtag.GenerateOptions{
+			Tag:                  revtag.DefaultRevisionName,
+			Revision:             rev,
+			Overwrite:            true,
+			AutoInjectNamespaces: autoInjectNamespaces,
+		}
+		// If tag cannot be created could be remote cluster install, don't fail out.
+		tagManifests, err := revtag.Generate(context.Background(), client, o, ns)
+		if err == nil {
+			if err = applyManifests(client, tagManifests); err != nil {
+				return false, err
+			}
+		}
+		processed = true
+	}
+	return processed, nil
+}
+
+func applyManifests(kubeClient kube.CLIClient, manifests string) error {
+	yamls := strings.Split(manifests, revtag.Separator)
+	for _, yml := range yamls {
+		if strings.TrimSpace(yml) == "" {
+			continue
+		}
+
+		obj := &unstructured.Unstructured{}
+
+		if err := yaml.Unmarshal([]byte(yml), obj); err != nil {
+			return fmt.Errorf("failed to unmarshal YAML: %w", err)
+		}
+		var gvr schema.GroupVersionResource
+		if obj.GetKind() == name.MutatingWebhookConfigurationStr {
+			gvr = schema.GroupVersionResource{
+				Group:    "admissionregistration.k8s.io",
+				Version:  "v1",
+				Resource: "mutatingwebhookconfigurations",
+			}
+		} else if obj.GetKind() == name.ValidatingWebhookConfigurationStr {
+			gvr = schema.GroupVersionResource{
+				Group:    "admissionregistration.k8s.io",
+				Version:  "v1",
+				Resource: "validatingwebhookconfigurations",
+			}
+		}
+
+		_, err := kubeClient.Dynamic().Resource(gvr).Namespace(obj.GetNamespace()).Patch(
+			context.TODO(), obj.GetName(), types.ApplyPatchType, []byte(yml), metav1.PatchOptions{
+				Force:        nil,
+				FieldManager: "install",
+			})
+		if err != nil {
+			return fmt.Errorf("failed to apply YAML: %w", err)
+		}
+	}
+	return nil
+}
+
+// operatorManageWebhooks returns .Values.global.operatorManageWebhooks from the Istio Operator.
+func operatorManageWebhooks(iop *iopv1alpha1.IstioOperator) bool {
+	if iop.Spec.GetValues() == nil {
+		return false
+	}
+	globalValues := iop.Spec.Values.AsMap()["global"]
+	global, ok := globalValues.(map[string]any)
+	if !ok {
+		return false
+	}
+	omw, ok := global["operatorManageWebhooks"].(bool)
+	if !ok {
+		return false
+	}
+	return omw
+}
+
+func processDefaultWebhookAfterReconcile(iop *iopv1alpha1.IstioOperator, client kube.Client) error {
+	var ns string
+	if configuredNamespace := iopv1alpha1.Namespace(iop.Spec); configuredNamespace != "" {
+		ns = configuredNamespace
+	} else {
+		ns = constants.IstioSystemNamespace
+	}
+	rc := client.RESTConfig()
+	// operator controller requires high privileges, so can use the CLIClient to apply yamls.
+	kubeClient, err := kube.NewCLIClient(kube.NewClientConfigForRestConfig(rc), "")
```

Comment:
I don't think this makes sense. Note `kube.Client` and `kube.CLIClient` are the same struct under the hood, just different interfaces to prevent things like Istiod from using functions it should not.

This code is actually making a new client, which is a new connection to api-server, etc. And possibly many times.

Probably we can just make it use a CLIClient by updating `Add` to make the new cli client instead.

Reply:
Thanks for reviewing this. I think `kube.CLIClient` is no longer needed here. Originally, it was used for applying tag webhook YAMLs with `ApplyYAMLFiles`. However, the operator is read-only, which means the creation of webhook yamls before applying will fail. As a result, I revised the process to use the client to create the configs. I'm going to revise the client usage.